### ISSUE TITLE
When opening a new adjacent terminal, focus is not always given to it

### ISSUE THREAD

**Issue by bagage** [gnunn1/terminix#525]
When opening a new terminal with `ctrl+alt+d` or `ctrl+alt+r`, the mouse focus is given to the newly created terminal.

However, when the *current* terminal is firing some output (eg. executing `make`, or `aptitude update`) the focus *may* be left in the current terminal. The more output, the more the bug has chance to happen.
It seems that's not the only way to reproduce the issue (I run into it from time to time), but that's the first way I found easily reproducible.

Running on master, Debian 4.7.6-1, Wayland, Gnome-shell 3.22.1. If you need additional information I'd gladly help!

**Comment by gnunn1** [comment]
I just tried it while building gtk+ and the new terminal always got the focus for me. I'm running Arch Linux with Wayland and gnome 3.22.1 as well so pretty similar stack. Now gtk+ chugs a bit when building so it isn't exactly flying up the screen, I'll play around some more and see if I can find a reproducer.

**Comment by bagage** [comment]
I am now even reproducing a similar bug when creating a new tab (`ctrl+t`): even if the cursor is moved to the new tab, it is not displayed. See [this webm](http://www.webm.land/w/FZyd/) which illustrates the bug.

**Comment by gnunn1** [comment]
I can reproduce in Wayland as well, I think this is a visual VTE bug. The terminal has focus and if you type it appears in the terminal just fine, however the cursor in the VTE remains an outlined block rather then the solid block you expect to see.

There was an upstream report here which marked this as fixed, however you can see folks still reporting it:

https://bugzilla.gnome.org/show_bug.cgi?id=677329

**Comment by gnunn1** [comment]
@bagage I've fixed the reproducer you posted. This was caused by my attempt to make #540 work. For that issue, the terminal needed to be added to the widget hierarchy before spawning the terminal. This resulted in multiple show calls and changes in the order of activities which I think is triggering this bug in the VTE. Re-working the sequence to follow what I had originally seems to have sorted this out.
